### DISCUSSION TITLE
ArrayTable 校验的时候跳过了新加的未渲染的行

### DISCUSSION THREAD

**Discussion by JE-lee** [alibaba/formily#3116] [Q&A 2.x] [answered]
直接在 [@formily/element](https://element.formilyjs.org/guide/array-table.html) 组件文档中操作可复现。
1. 在第一页连续添加 11条数据，此时**不要切换到第 2 页**。
2. 将第 1 页的数据全部填写完整
3. 
![image](https://user-images.githubusercontent.com/19794813/167983910-cb046f53-9855-4823-a6c2-cd8c39e57476.png)

4. 点击提交
5.  此时查看 console， 发现第 11 行数据为空，但是依然校验通过
![image](https://user-images.githubusercontent.com/19794813/167984379-0522ed9e-a100-41df-8340-0b5bc2816d9d.png)

这个应该也算 bug 吧。
好像是 如果不切换到第2页面，对应的 `<RecursionField>` 不会渲染，也就不会调用 Form.createField，Form 里面就没有对应的 Field, 检验的时候就跳过了

**Comment by janryWang** [comment, marked as answer]
> 好像是 如果不切换到第2页面，对应的 <RecursionField> 不会渲染，也就不会调用 Form.createField，Form 里面就没有对应的 Field, 检验的时候就跳过了

说的对，组件应该改一下，添加超过一页的数据，自动切换下一页，欢迎PR

**Reply by JE-lee** [reply]
给 @formily/element, antd, next 提了 [PR](https://github.com/alibaba/formily/pull/3117), 实现的就是上面的逻辑
